src/config.rs

```diff
@@ -1542,33 +1565,83 @@ impl DBConfigManger {
                  block-cache.capacity in storage module instead"
                 .into());
         }
-        let opt = self.db.get_options_cf(cf)?;
-        opt.set_block_cache_capacity(size.0)?;
+        let mut result = Ok(());
+        self.tablet_factory
+            .loop_tablet_cache(Box::new(|region_id, suffix, db: &RocksEngine| {
+                let r = db.get_options_cf(cf);
+                if let Ok(opt) = r {
+                    let r = opt.set_block_cache_capacity(size.0);
+                    if r.is_err() {
+                        result = Err(Box::from(r.err().unwrap()));
+                        error!(
+                            "opt.set_block_cache_capacity on cf {} with size {}MB failed on tablet {}_{}. err:{:?}",
+                            cf,
+                            size.as_mb(),
+                            region_id,
+                            suffix,
+                            &result
+                        );
+                    }
+                } else {
+                    result = Err(Box::from(r.err().unwrap()));
+                    error!(
+                        "get_options_cf {} failed on tablet {}_{}. err:{:?}",
+                        cf, region_id, suffix, &result
+                    );
+                }
+            }));
         // Write config to metric
         CONFIG_ROCKSDB_GAUGE
             .with_label_values(&[cf, "block_cache_size"])
             .set(size.0 as f64);
-        Ok(())
+        result
     }
 
     fn set_rate_bytes_per_sec(&self, rate_bytes_per_sec: i64) -> Result<(), Box<dyn Error>> {
-        let mut opt = self.db.as_inner().get_db_options();
-        opt.set_rate_bytes_per_sec(rate_bytes_per_sec)?;
-        Ok(())
+        let mut result = Ok(());
+        self.tablet_factory
+            .loop_tablet_cache(Box::new(|region_id, suffix, db: &RocksEngine| {
```

Comment:
rate limiter should be shared, so this is unnecessary to loop.

Reply:
The current implementation of rate limiter is not shared. It's per DB. Once it's changed to the shared one, then we may change the code here as well.

Reply:
Rate limiter should be shared, every db has its own rate limiter actually means no limit.

Reply:
Shared rate limiter API is not ready in rust-rocks. I'd like to postpone the changes to rust-rocks due to the development of tirocks.